### PR TITLE
[NPM-3110] Fix revive lint warnings for the networks team

### PR DESCRIPTION
**Pull request by pimlu**
### What does this PR do?

This fixes revive linter warnings for the Networks team - the revive linter was recently enabled but to decouple it with each team, it was done by adding linter suppressors on the offending lines. This PR does our part by deleting those suppressors and fixing the warnings.

### Additional Notes

A small part is files owned by the process team which are noted in a comment below.

### Describe how to test/QA your changes

Mainly, ran the linters on linux:
`inv -e show-linters-issues --filter-team "@DataDog/Networks" --filter-linters "revive" --platforms "linux,arm64"`
But also ran them on MacOS because there were some build-tag dependent files:
`inv -e show-linters-issues --filter-team "@DataDog/Networks" --filter-linters "revive" --platforms "darwin,arm64"`
However, I also ran into some additional revive issues for some reason these two did not catch during the `git push` hook. I could rerun that linter like this:
`inv linter.go --only-modified-packages`

### CODE REVIEW

File: pkg/process/checks/net.go [hunk 1]
```diff
@@ -408,11 +408,10 @@ func batchConnections(
 				continue
 			}
 
-			//nolint:revive // TODO(NET) Fix revive linter
```
Comment (pimlu) [inline pkg/process/checks/net.go:411]: These files are Networks related but have Process team code ownership which is why Processes got pulled into code review

File: pkg/network/tracer/offsetguess/tracer.go [hunk 2]
```diff
@@ -677,7 +675,7 @@ func (t *tracerOffsetGuesser) checkAndUpdateCurrentOffset(mp *maps.GenericMap[ui
 			if !t.guessTCPv6 && !t.guessUDPv6 {
 				t.logAndAdvance(t.status.Offset_sk_buff_head, GuessNotApplicable)
 				return t.setReadyState(mp)
-			} else { //nolint:revive // TODO(NET) Fix revive linter
+			} else {
```
Comment (hmahmood) [inline pkg/network/tracer/offsetguess/tracer.go:678]: Shouldn't the `else` be removed?

**Review by akarpz — APPROVED** [review]
lgtm aside from hasan's comment

File: pkg/network/tracer/offsetguess/tracer.go (hunk 2 above)
Comment (pimlu) [reply pkg/network/tracer/offsetguess/tracer.go:678]: Yeah now that you mention it I see this was probably a return-else linter rule. It sure seems something was different about the linting methodology in the original PR cause I see no lint error here at all when I run it. There are multiple ways to invoke the linter so maybe they were using it differently than me.

**Review by daniel-taf — APPROVED** [review]

**Pull request merged**